### PR TITLE
Does my coding partner follow JS best practices?

### PR DESCRIPTION
**Pull request by udeaghad**
Kindly review my codes

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    <link rel="stylesheet" href="style.css">
+    <title>To-do-list Exercise</title>
+</head>
+<body>
+ <main>
+<section>
+  <div class="heading">
+    <h1>Today's To Do</h1>
+    <i class="fa-solid fa-rotate"></i>
+  </div>
```
Comment (sajid-munawar) [inline index.html]: - Please wrap your heading into a header as it would be better to use more semantic HTML tags.

Comment (sajid-munawar) [inline index.html]: - Please wrap your heading into a header as it would be better to use more semantic HTML tags.

File: index.html [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    
+    
+  </div>
+    <div class="to-do-cont" id="to-do-cont"></div>
+    <button type="button" id="clear-btn" class="clear-btn">Clear all completed</button>
+    
+
+
+
+
```
Comment (sajid-munawar) [inline index.html]: - Please be careful with blank lines and indentation. Do not add blank lines, spaces, or indentations without a reason.

Comment (sajid-munawar) [inline index.html]: - Please be careful with blank lines and indentation. Do not add blank lines, spaces, or indentations without a reason.

File: index.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+  });
+
+  numberIndex();
+};
+
+// add event listerner to the imported checkItem to perform their functions
+container.addEventListener('click', checkItem);
+
+// add event listener to the input to create items added on the addItem function
+inputAdd.addEventListener('keypress', (e) => {
```
Comment (sajid-munawar) [inline index.js]: - You can remove parenthesis in an arrow function if there is a single argument.

Comment (sajid-munawar) [inline index.js]: - You can remove parenthesis in an arrow function if there is a single argument.

**Review by sajid-munawar — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @udeaghad  👋, 

Good job so far!
There are some issues that you still need to work for :
[JavaScript best practices](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).
[DRY, KISS, and YAGNI](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/clean-code/lesson_DRY_KISS_YAGNI.md) rules.
[HTML and CSS best practices](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md).
## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- GitHub flow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

File: index.html (hunk 2 above)
Comment (udeaghad) [reply index.html]: @sajid-munawar 

Changes implemented

File: index.js (hunk 3 above)
Comment (udeaghad) [reply index.js]: @sajid-munawar 

Changes implemented.

File: index.html (hunk 1 above)
Comment (udeaghad) [reply index.html]: @sajid-munawar 

Changes implemented.

**Review by sajid-munawar — APPROVED** [review]
## STATUS: Approved ♻️ 

Hi @udeaghad 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Pull request merged**